### ISSUE TITLE
[30k]: Open source onboarding

### ISSUE THREAD

**Issue by widal001** [HHS/simpler-grants-gov#72]
### Key links

- [Deliverable specification](https://simpler-grants.gitbook.io/wiki/product/deliverables/specifications/open-source-onboarding)
- [Simpler Grants.gov full roadmap](https://github.com/orgs/HHS/projects/12)
- [10k milestones](https://github.com/orgs/HHS/projects/13/views/11?sliceBy%5Bvalue%5D=Open+source+onboarding)

### Description

- **What:** Set up the tools and processes needed to build an open source community around the Simpler Grants.gov initiative
- **Why:** Ensures that open source contributors and the general public can easily participate in the project and provide input and code
- **Who:**
  - Open source contributors
  - Other external stakeholders

> [!NOTE]
> For more information about this deliverable, please refer to the deliverable specification linked above.

**Comment by EOKENAVA** [comment]
**Weekly Update**

What has been done?


- Reviewed the security policies of these pieces of software and ensure that they already say that email is shared with the admin of the tool (us)
- Reviewed the terms of service for data collection of the apps (Google Groups, Zoom, Slack) 
- Developed and documented team norms for reviewing public code contributions to ensure consistency and quality.

Status
On Track

Risk
None

Deliverable Date
30k deliverable: mid-Sprint 20 - 4/02

What will be done?
Detail the community engagement processes including meetings, attendance expectations, and cadence within the wiki.

**Comment by EOKENAVA** [comment]
**Weekly Update**

**What has been done?**
- Detailed the community engagement processes including meetings, attendance expectations, and cadence within the wiki.
- Developed a clear onboarding process for public users to access and use development tools.

**Status**
On Track

**Risk**
None

**Deliverable Date**
30k deliverable: mid-Sprint 20 - 4/02

**What will be done?**
- Develop a detailed onboarding process for Slack, including instructions for creating channels, and document it on the wiki.
- Create comprehensive documentation for internal staff on public policies, including guidelines on what to post and where.
- Document the onboarding process for GitBook, making it easy for new contributors to start using the platform.

**Comment by EOKENAVA** [comment]
**Weekly Update**

**What has been done?**
- Developed a detailed onboarding process for Slack, including instructions for creating channels, and document it on the wiki.
- Created comprehensive documentation for internal staff on public policies, including guidelines on what to post and where.
- Documented the onboarding process for Github, making it easy for new contributors to start using the platform.
- Developed an integration to receive notifications of GitBooks Wiki updates in Slack Chat
- Detailed the community engagement processes including meetings, attendance expectations, and cadence within the wiki.
- Developed an integration between GitHub and GitBook to automate the synchronization of content, facilitating seamless updates and collaboration on the public wiki.

**Status**
On Track

**Risk**
None

**Deliverable Date**
30k deliverable: mid-Sprint 20 - 4/02

**What will be done?**
- Implement an integration to display Google Groups Calendar events within the GitBooks Wiki.
- Enable an integration that automatically updates the wiki with Zoom meeting details.
- Implement an integration that enables receiving updates on GitHub tickets directly in Slack.

Currently _Blocked_ on developing a clear onboarding process for public users to access and use development tools.

**Comment by btabaska** [comment]
Currently awaiting guidance on usage of Google Forms to collect and store open source developer data. @lucasmbrown-usds said that they should get back to us by Friday. Use of this tool is blocking work because we need to know where and how we will be able to collect and store data on developers who want to sign up to be a part of the community. We selected Google Forms after a conversation with Remy DeCausemaker as the preferred tool for this.

**Comment by sarahknoppA6** [comment]
Awesome, thank you Brandon!!

**Comment by margaretspring** [comment]
**Weekly Update**

**What has been done?**
- Developed a detailed onboarding process for Slack, including instructions for creating channels, and document it on the wiki.
- Created comprehensive documentation for internal staff on public policies, including guidelines on what to post and where.
- Documented the onboarding process for Github, making it easy for new contributors to start using the platform.
- Developed an integration to receive notifications of GitBooks Wiki updates in Slack Chat
- Detailed the community engagement processes including meetings, attendance expectations, and cadence within the wiki.
- Developed an integration between GitHub and GitBook to automate the synchronization of content, facilitating seamless updates and collaboration on the public wiki.
- Gained approval to move ahead with Google Forms for initial outreach with community. 

**Status**
On Track

**Risk**
None

**Deliverable Date**
30k deliverable: mid-Sprint 20 - 4/02

**What will be done?**
- Implement an integration to display Google Groups Calendar events within the GitBooks Wiki.
- Enable an integration that automatically updates the wiki with Zoom meeting details.
- Implement an integration that enables receiving updates on GitHub tickets directly in Slack.

**Comment by margaretspring** [comment]
**Weekly Update - IN DRAFT**

**What has been done?**
-ddgf
-

**Status**


**Risk**


**Deliverable Date**


**What will be done?**
-df

**Comment by margaretspring** [comment]
@sarahknoppA6 - Who can address the question about the labeling that I asked above?

**Comment by sarahknoppA6** [comment]
@margaretspring I'm not able to open the link you provided. But maybe @widal001 can take a look at this- do you know which pieces of it are out of sync? 
I don't think we change the names of the 30ks once they start because it impacts the wiki URL and we don't want to change those- I'm not sure why there's an inconsistency- Billy do you know?

**Comment by widal001** [comment]
> This 30k description is out of sync with the [Spec on GitBook](https://simpler-grants.gitbook.io/public-wiki/9bkdlmwZQsFh9t1xkhc5/product/deliverables/specifications/open-source-onboarding). Nava's PoP deliverable title also Open Source Group Kickoff. Can it be clarified/reconciled here?

@margaretspring The name was changed to "Open source onboarding" from "Open source group kickoff" as part of some discussions that Lucas, Sumi, and Aaron and I had in which we updated several of the 30k names. Part of the motivation for changing this to "onboarding" from "kickoff" was that we had descoped setting up regular meetings, and instead focused on the open source tool onboarding process.

And with regards to the description being out of sync would you mind highlight which parts seemed inconsistent? As far as I could tell the following matched:
- Summary
- Business goals
- Definition of done (with some small formatting discrepancies -- it looks like the first checklist item wasn't copied over as a checklist item and didn't get indented, but the criteria itself still seemed to match)

**Comment by sarahknoppA6** [comment]
We confirmed at ToT last week that completion of this 30k is on track for tomorrow. I was looking through the associated 10ks and it looks like there are several tasks still outstanding. (I do understand that tasks associated with procurement and the public wiki domain link may lag based on previous updates.) I'm not able to easily tell if these just need to be updated, are no longer relevant, or will be completed. Would you mind re-confirming this delivery date, @sumiat and @margaretspring ?

I'm working on my slides for OG leadership today and want to confirm we're expecting this 30k to be finalized tomorrow. Time permitting, it may be helpful to have a meeting where we review the 30k together and close it out at the onsite on Thursday. 
cc @lucasmbrown-usds

**Comment by btabaska** [comment]
This 30k will not be done by tomorrow due to blockers that are somewhat outside our control. It will take 1 more sprint to complete.

Unfinished Items with reasoning of delay:

- **Zoom** -> Procurement is still in process. We will use Slack Group Calls to work around this delay
- **Services are accessible to all people on the HHS network, public internet, and preferably the White House network and most or all agency networks** -> This is part of the definition of done for onboarding 3 people. Will complete with that task
-  **Instructions for the internal team that assists with onboarding new users are clearly and accessibly documented in our public or internal wiki** -> This will be done by EOD tomorrow. Just needs review
- **The onboarding page is linked from simpler.grants.gov** -> We are experiencing technical delays on getting the custom domain for wiki.simpler.grants.gov. I am working with Cloudflare, Terrance, and GitBook to try and resolve this.
- **The onboarding page is linked from the simpler-grants-gov GitHub repo** - Same as above
- **We have onboarded three members of the general public to the following communication tools:** Lucas and I have had scheduling conflicts to determining the initial cohort of users to onboard. I will make sure to schedule time with them and get the initial users onboarded by end of Sprint 21
- **The public wiki can be accessed from a custom domain that is easy to remember and specific to the Simpler.Grants.gov initiative** -> See above technical delay explanation
- **Public Slack channels have been reviewed and sensitive content has been removed or the channel has been made private** -> Sumi took on this task and has been running into delays. Should be complete by end of sprint 21
- **At least three GitHub issues have been labeled with "help wanted"** -Tickets have been identified, but will be flagged once we have users to work them so that we do not delay active eng work. Will be done by end of sprint 21
-  **We've documented an information architecture for the public wiki, so that content has clear organizing principles and it is easy to know where to add or look for content** -> This was moved from the initial definition of done for the epic since it won't impact getting users involved in the open source group.
- **We have a system for "offboarding" members of the community from all tools** -> This will be complete by end of sprint 20, it just needs to be marked for review.
- **We are able to "block" users who violate community guidelines and offboard them from all tools without their participation in the offboarding process** -> This will be complete by end of sprint 20, it just needs to be marked for review.

To summarize:
**Total Remaining Tickets:**
11
**Tickets Blocked:**
1 - Due to procurement of Zoom
4 - Due to ongoing wiki.simpler.grants.gov custom domain troubleshooting
**Tickets that will be done by end of sprint 20**
3 - In Review
**Tickets that will need 1 more sprint to complete**
3 - Needs more time to complete

**Comment by btabaska** [comment]
Thanks Lucas for the response, reading through the feedback there is a lot that can be improved on going forward. I apologize for getting desynced on this body of work.

For item 1: This sounds good, I'll reach out so we can chat about the process for setting up these calls.

For item 2: I believe that this may have been a miscommunication on my part. Currently the wiki is still blocked with the custom domain, so as soon as this is unblocked I'll reach out to determine a good set of users to verify that our tools are accessible on those networks.

For item 6: I apologize, I did not want to paint you as a blocker. I may have worded that poorly. I'll reach out to you async and we can chat. My understanding is that you had a cohort of users that you were interested in onboarding. 

On item 9: again, I worded this poorly. Aaron and I will work on creating a list of marked work. I was worried about marking tickets that were going to be picked up by internal developers before an open source developer would be onboarded.

On item 10: I could have worded this better. It will not be descoped from the epic. It will be pushed back to after we are onboarding open source contributors since we want to make that happen as soon as possible. 

Again, I appreciate the responses and we can chat more about any of these items. I was trying to be as succinct as possible while remaining thorough and accountable to the work and I think that some of my wording was a bit off. Thanks for the understanding from everyone.

**Comment by margaretspring** [comment]
**Weekly Update**

**What has been done?**

- Developed a detailed offboarding process for community members who no longer wish to be involved. Documented in the private wiki.
- Created comprehensive documentation for internal staff on escalated offboarding, including instruction on tool by tool offboarding, documentation of the event, and a full guide on steps to follow if a community members breaks the code of conduct.
- Began process of setting public wiki up with a custom domain.
- Documented Instructions for the internal team in the private wiki that assists with onboarding new users are clearly and accessibly documented in our public or internal wiki

**Status**  
Delayed

**Risk**  
_Blocked_ Community Zoom Tool setup - due to procurement issues with the license. Lucas is working on getting us another existing license. IF we are unable to get a Zoom license THEN we will use Lucas' secondary license or use Slack video conferencing

_Blocked_ - We are experiencing technical delays on getting the custom domain for [wiki.simpler.grants.gov](http://wiki.simpler.grants.gov/). I am working with Cloudflare, Terrance, and GitBook to try and resolve this. Blocked work includes:
- The onboarding page is linked from the simpler-grants-gov GitHub repo
- The public wiki can be accessed from a custom domain that is easy to remember and specific to the [Simpler.Grants.gov](http://simpler.grants.gov/) initiative
- The onboarding page is linked from [simpler.grants.gov](http://simpler.grants.gov/) 
IF we are unable to get a working custom domain THEN we will proceed with onboarding the first few people manually, but we will need to share documentation in a really hacky way since there won't be a good static public link to the wiki

**Deliverable Date**  
30k deliverable: end-Sprint 21 - 4/14

**What will be done?**

- Verify as a definition of done that services are accessible to all people on the HHS network, public internet, and preferably the White House network and most or all agency networks
- Onboarding three members of the general public to the following communication tools
- Public Slack channels have been reviewed and sensitive content has been removed or the channel has been made private
- At least three GitHub issues have been labeled with "help wanted"
- We've documented an information architecture for the public wiki, so that content has clear organizing principles and it is easy to know where to add or look for content

**Comment by btabaska** [comment]
**What has been done?**

- Developed a detailed training for policies around PII and sensitive data in public Slack channels. Shared out with the team
- Completed scan of all public Slack channels for PII and sensitive information. Informed posters of that information of it being in slack and requested its removal. 
- Continued process of setting up custom domain for Wiki. Currently taking longer than expected so I increased the amount of points on the ticket. Currently working with GitBook & HHS support staff to uncover why cloudflare is blocking the DNS.
- Reached out to the initial 3 users to be onboarded to the open source group. Currently have acceptance from 2/3 and will begin onboarding by the end of this week.
- Marked 3 "help wanted" issues on github for the initial cohort of open source developers to work on.

**Status**  
Delayed

**Risk**  
_Blocked_ Community Zoom Tool setup - due to procurement issues with the license. Lucas is working on getting us another existing license. IF we are unable to get a Zoom license THEN we will use Lucas' secondary license or use Slack video conferencing

_Blocked_ - We are experiencing technical delays on getting the custom domain for [wiki.simpler.grants.gov](http://wiki.simpler.grants.gov/). I am working with Cloudflare, Terrance, and GitBook to try and resolve this. Blocked work includes:

- The onboarding page is linked from the simpler-grants-gov GitHub repo
- The public wiki can be accessed from a custom domain that is easy to remember and specific to the [Simpler.Grants.gov](http://simpler.grants.gov/) initiative
- The onboarding page is linked from [simpler.grants.gov](http://simpler.grants.gov/)  
    IF we are unable to get a working custom domain THEN we will proceed with onboarding the first few people using a working hhs.gov domain.

**Deliverable Date**  
30k deliverable: end-Sprint 21 - 4/14

**What will be done?**

- Verify as a definition of done that services are accessible to all people on the HHS network, public internet, and preferably the White House network and most or all agency networks
- Onboarding three members of the general public to the following communication tools
- We've documented an information architecture for the public wiki, so that content has clear organizing principles and it is easy to know where to add or look for content
- Close out 30k for Open Source Onboarding, move into coding challenge 30k, plan for recruitment of open source developers

**Comment by widal001** [comment]
Updated the format of this ticket to reduce discrepancies with the deliverable spec.

**Comment by margaretspring** [comment]
**Weekly Update 4/17**

**What has been done?**


**Status**


**Risk**
  

**Deliverable Date**


**What will be done?**

**Comment by btabaska** [comment]
Weekly Update 4/17

**What has been done?**

- Migrated Wiki to custom domain
- Created Training on Public sensitive data policies and shared with SGG Staff
- Scanned and removed sensitive PII information from public slack channels prior to onboarding public
- Created link to SGG Public wiki from the Simpler Grants Website 
- Linked the SGG Public wiki from the Simpler Grants GitHub Repo's README.md
- Set up custom domain "wiki.simpler.hss.gov" for public wiki to point to
- Onboarded 1 individual to Open Source Community
- Procured Zoom License as of this morning so we can host meetings with the public Open Source group
- Sent invites to onboard to a few selected open source community members to be a part of the initial cohort

**Status**
Delayed

**Risk**
Blocked - We are experiencing technical delays on getting the custom domain for [wiki.simpler.grants.gov](http://wiki.simpler.grants.gov/). I am working with Cloudflare, Terrance, and GitBook to try and resolve this. Blocked work includes:
IF we are unable to make the grants.gov domain work, THEN we can use the hhs.gov domain that we are using now.

Blocked - We have reached out to 8 individuals to be a part of the initial Open Source Community for SGG. Of those 8, only 5 agreed to be a part of the community at this time. We sent them onboarding instructions and are awaiting them joining. We have gotten 1 fully onboarded and are awaiting more folks to complete the process.

**Deliverable** Date
30k deliverable: beginning-Sprint 22 - 4/19

**What will be done?**

- Complete onboarding of remaining 2 Open Source Group Members

**Comment by btabaska** [comment]
Weekly Update 4/24

What has been done?
Migrated wiki from custom domain to grants.gov domain
Met with GitBook to resolve issues and provide feedback
Met goal of onboarding 3 contributors 

Status
Complete

Risk
None

Deliverable Date
30k deliverable: beginning-Sprint 22 - 4/19

What will be done?

Complete

**Comment by widal001** [comment]
This was completed last week after we onboarded the first set of open source user!
